drivers/flash/spi_nor.c

```diff
@@ -381,6 +382,9 @@ static int spi_nor_write(struct device *dev, off_t addr, const void *src,
 static int spi_nor_erase(struct device *dev, off_t addr, size_t size)
 {
 	const struct spi_nor_config *params = dev->config->config_info;
+	const u32_t *erase_size = params->erase_size;
+	const u8_t *erase_cmd = params->erase_cmd;
+	int idx = 0;
```

Comment:
I think this needs to move into the `while` at the top, otherwise if `size` requires multiple erase operations things will behave very badly.

Reply:
You're right. My bad. I also moved the call for write enable because in some cases the code sets it without a following erase command. The same goes for the spi_nor_wait_until_ready call.